src/JMS/Serializer/GraphNavigator.php

```diff
@@ -210,7 +210,13 @@ public function accept($data, array $type = null, Context $context)
                 }
 
                 if ($context instanceof DeserializationContext && ! empty($metadata->discriminatorMap) && $type['name'] === $metadata->discriminatorBaseClass) {
-                    $metadata = $this->resolveMetadata($data, $metadata);
+                    try {
+                        $metadata = $this->resolveMetadata($data, $metadata);
+                    } catch(\LogicException $e) {
+                        if($metadata->discriminatorStrictDeserialize) {
```

Comment:
what will be the content of `$metadata` if the `LogicException` is thrown? the abstract class defined on `@Type`?